src/jdk.incubator.code/share/classes/jdk/incubator/code/writer/OpWriter.java

```diff
@@ -557,7 +558,7 @@ <T> void writeSeparatedList(String separator, Iterable<T> l, Consumer<T> c) {
     }
 
     void writeType(TypeElement te) {
-        write(te.externalize().toString());
+        write(JavaTypeUtils.flatten(te.externalize()).toString());
```

Comment:
When writing a type element into textual form, flatten it first (to make it more readable if it's a Java type).